### ISSUE TITLE
umount: can't umount /Users: Invalid argument mount.nfs: an incorrect mount option was specified

### ISSUE THREAD

**Issue by lukebarton** [codekitchen/dinghy#78]
Dinghy 4.0.2 - Brew says it's stable branch but i'm pretty sure I reinstalled with latest from prerelease rather than the release trying to address #69 

$ dinghy up
Starting the dinghy VM...

Stopping NFS daemon, this will require sudo
Password:
Starting NFS daemon, this will require sudo
Waiting for NFS daemon...
Mounting NFS /Users/lukebarton
umount: can't umount /Users: Invalid argument
mount.nfs: an incorrect mount option was specified
mount: mounting 192.168.99.1:/Users/lukebarton on /Users/lukebarton failed: Device or resource busy
/usr/local/Cellar/dinghy/HEAD/cli/dinghy/ssh.rb:23:in `run': Error executing command: ["sudo mount -t nfs 192.168.99.1:/Users/lukebarton /Users/lukebarton -o nfsvers=3,udp,mountport=19321,port=19321,nolock,hard,intr"] (Ssh::CommandFailed)
        from /usr/local/Cellar/dinghy/HEAD/cli/dinghy/machine.rb:76:in `ssh'
        from /usr/local/Cellar/dinghy/HEAD/cli/dinghy/machine.rb:72:in `mount'
        from /usr/local/Cellar/dinghy/HEAD/cli/cli.rb:181:in `start_services'
        from /usr/local/Cellar/dinghy/HEAD/cli/cli.rb:78:in `up'
        from /usr/local/Cellar/dinghy/HEAD/cli/thor/lib/thor/command.rb:27:in `run'
        from /usr/local/Cellar/dinghy/HEAD/cli/thor/lib/thor/invocation.rb:126:in `invoke_command'
        from /usr/local/Cellar/dinghy/HEAD/cli/thor/lib/thor.rb:359:in `dispatch'
        from /usr/local/Cellar/dinghy/HEAD/cli/thor/lib/thor/base.rb:440:in `start'
        from /usr/local/bin/_dinghy_command:6:in `<main>'

**Comment by codekitchen** [comment]
Similar to #69 , I would suspect a race condition here. There is boot2docker/docker-machine code to mount `/Users` using vbox file sharing during VM startup, I bet the dinghy code to unmount `/Users` so that we can mount the NFS share instead is running too soon.

**Comment by robertcapasso** [comment]
I modified the command run by the startup script to only umount /Users if it shows up in /proc/mounts:

```ruby
ssh("if [ $(grep -c '\/Users' /proc/mounts) -gt 0 ]; then sudo umount /Users; fi;")
```

**Comment by MrMMorris** [comment]
I hit this issue every once in a while if the VM or my macbook goes to sleep. I usually have to `dinghy halt && dinghy up` a couple times to get it going again.

This PR would be nice: https://github.com/codekitchen/dinghy/pull/98

**Comment by codekitchen** [comment]
I've merged #98 to master, which will remove the `umount: can't umount /Users: Invalid argument` error, but that error was being ignored. I don't think the fatal error on the next line, `mount.nfs: an incorrect mount option was specified`, is fixed yet.

**Comment by codekitchen** [comment]
I'm going to close this since we haven't seen any reports in a long time. Hopefully it's fixed, if not we can open a new issue.
